OpenTrons/opentrons-api
# feat(protocol-designer): create migration for PD 8.2.0

# Overview

This PR creates the initial migration from PD 8.1.0 -> 8.2.0. Specifically, I create the 8.2.0 migration file and wire it up in the migration index file. The scope of this initial migration pass is to update the handling of time fields for pause and heater shaker step forms. At import, we consolidate the individual time unit fields for these forms into a single time field. I also update all downstream utilities, types, defaults, and tests that were affected.

Closes AUTH-675
Closes AUTH-760
Closes AUTH-859
Closes AUTH-871
Closes AUTH-1000

## Test Plan and Hands on Testing

- Create protocol in production PD that contains a heater shaker form with a timer and a pause form that will wait for a specified duration ([example](https://github.com/user-attachments/files/17621673/migrate.json.zip))
- Export the protocol and import in new PD built off this branch
- Verify that saved step forms populate the timer fields appropriately

## Changelog



## Review requests



## Risk assessment

medium
